aplib.net-demo/Assets/Testing/AplibTests/GameplayTests.cs

```diff
@@ -0,0 +1,879 @@
+                0.3f);
+            TransformPathfinderAction moveToVisibleKeyAction = new(
+                beliefSet => beliefSet.PlayerRigidBody,
+                beliefSet => beliefSet.VisibleKeys.Observation
+                    .Select(x => x.item.transform.position)
+                    .OrderBy(x => Vector3.Distance(x, beliefSet.PlayerRigidBody.Observation.position))
+                    .First(),
+                0.3f);
+
+            PrimitiveTactic pickUpVisibleHealthPotion = new(moveToVisibleHealthPotionAction,
+                beliefSet => beliefSet.VisibleHealthPotions.Observation.Length != 0);
+            PrimitiveTactic pickUpVisibleRagePotion = new(moveToVisibleRagePotionAction,
+                beliefSet => beliefSet.VisibleRagePotions.Observation.Length != 0);
+            PrimitiveTactic pickUpVisibleAmmo = new(moveToVisibleAmmoAction,
+                beliefSet => beliefSet.VisibleAmmo.Observation.Length != 0);
+            PrimitiveTactic pickUpVisibleKey = new(moveToVisibleKeyAction,
+                beliefSet => beliefSet.VisibleKeys.Observation.Length != 0);
+
+            Tactic fetchPotionIfDistancedMelee = FirstOf(guard: beliefSet =>
+                    beliefSet.DetermineEnemyToFocus(out float distance) is MeleeEnemy && distance > 5 &&
+                    ((!beliefSet.Inventory.Observation.ContainsItem<RagePotion>() &&
+                      beliefSet.VisibleRagePotions.Observation.Length != 0) ||
+                     (beliefSet.PlayerHealthPercentage.Observation < 30 &&
+                      beliefSet.VisibleHealthPotions.Observation.Length != 0)),
+                pickUpVisibleHealthPotion, pickUpVisibleRagePotion);
+
+            Tactic fetchVisibleItem = FirstOf(pickUpVisibleHealthPotion, pickUpVisibleRagePotion, pickUpVisibleAmmo, pickUpVisibleKey);
+            GoalStructure fetchVisibleItemGoal = new Goal(fetchVisibleItem, beliefSet => !beliefSet.AnyItemIsVisible);
+
+            #endregion
+
+            #region reactToEnemy
+
+            Action stepAsideRightAction = new(beliefSet =>
+            {
+                Rigidbody playerRigidBody = beliefSet.PlayerRigidBody;
+                playerRigidBody.AddForce(playerRigidBody.transform.right * 10, ForceMode.VelocityChange);
+            });
+            Action stepAsideLeftAction = new(beliefSet =>
+            {
+                Rigidbody playerRigidBody = beliefSet.PlayerRigidBody;
+                playerRigidBody.AddForce(playerRigidBody.transform.right * -10, ForceMode.VelocityChange);
+            });
+            PrimitiveTactic stepAsideLeft = new(stepAsideLeftAction, beliefSet =>
+            {
+                Rigidbody playerRigidBody = beliefSet.PlayerRigidBody;
+
+                Vector3 left = playerRigidBody.transform.TransformDirection(Vector3.left);
+                return Physics.Raycast(playerRigidBody.transform.position, left, 0.5f,
+                    ~LayerMask.GetMask("PlayerSelf", "Item", "Ignore Raycast"));
```

Comment:
It is not clear to me why you only need those layers. Maybe add a small comment explaining why